Src/Base/AMReX_Algorithm.H

```diff
@@ -157,6 +157,58 @@ namespace amrex
         return hi;
     }
 
+    template<typename ItType, typename ValType>
+    AMREX_GPU_DEVICE
+    ItType upper_bound (ItType first, ItType last, const ValType& val)
+    {
+#if AMREX_DEVICE_COMPILE
+        ItType it;
+        size_t count, step;
```

Comment:
I think it might not be safe to use `size_t`.  I think we should use `std::ptrdiff_t`.

Reply:
sure!